python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -465,25 +465,32 @@ def cutout_from_exposure(src_image, ra, dec, width, height, unit="pixel"):
             # pixel scale is defined as Angle/pixel
             wcs = src_image.getWcs()
             ps = wcs.getPixelScale().asArcseconds()
-            if ps != 0:
-                width = width / ps
-                height = height / ps
-            else:
-                raise Exception("Unexpected: pixel scale = 0")
+            width = width / ps
+            height = height / ps
         center = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
         size = afwGeom.Extent2I(width, height)
         cutout = src_image.getCutout(center, size)
         return cutout
 
-    def _cutout_by_data_id(self, data_id, ra, dec, width, height, unit="pixel"):
-        # check to see if q_results is empty
-        if data_id is None:
-            return None
+    def _cutout_by_data_id(self, data_id, ra, dec, width, height, unit="pixel",
+                           wcs=None):
+        # first need to check the cutout dimensions, by degree conversion
+        if unit in ["pixel", "pix", "px"]:
+            if not wcs:
+                metadata = self._metadata_from_data_id(data_id)
+                wcs = afwGeom.makeSkyWcs(metadata, strip=False)
+            ps = wcs.getPixelScale().asDegrees()
+            cutout_area = width * height * ps**2
+        else:  # arcsec as unit
+            cutout_area = width * height / 3600**2
+        if cutout_area > imgserv_config.MAX_IMAGE_CUTOUT_SIZE:
+            msg = f"Requested image exceeded " \
```

Comment:
You still might want to say what cutout_area is in the error message here, the max is of course constant, but knowing how much was requested can sometimes be very helpful in seeing typos